src/listeners.ts

```diff
@@ -0,0 +1,582 @@
+/**
+ * ULabel listener utilities.
+ *
+ * These primarily use JQuery, within the namespace "ulabel".
+ * Selectors and `.on` calls are intentionally kept in `create_ulabel_listeners`.
+ * Long handlers are broken out into separate functions.
+ */
+
+import type { ULabel } from "..";
```

Comment:
`import type` important to avoid circular import issues. You'll get a vague `webpack` import error otherwise